modules/bloxone_infra_host_gcp/main.tf

```diff
@@ -0,0 +1,145 @@
+/**
+ * # Terraform Module to create BloxOne Host in GCP
+ *
+ * This module will provision a GCP virtual machine that uses a BloxOne image.
+ * The virtual machine will be configured to join a BloxOne Cloud Services Platform (CSP) with the provided join token.
+ * If a join token is not provided, a new one will be created.
+ *
+ * The BloxOne Host created in the CSP is created automatically, and cannot be managed through terraform.
+ * A `bloxone_infra_hosts` data source is provided to retrieve the host information from the CSP.
+ * The data source will use the `tags` variable to filter the hosts.
+ * A `tf_module_host_id` tag will be added to the tags variable so that the data source can uniquely find the host.
+ *
+ * This module will also create a BloxOne Infra Service for each service type provided in the `services` variable.
+ * The service will be named `<service_type>_<host_display_name>`.
+ *
+/**
+ * ## Example Usage
+ *
+ * ```hcl
+ * module "bloxone_infra_host_gcp" {
+ *   source = "github.com/infobloxopen/terraform-provider-bloxone//modules/bloxone_infra_host_gcp"
+ *
+ *   name         = "bloxone-test-instance"
+ *   project_id   = "gcp-engineering"
+ *   region       = "us-east1"
+ *   zone         = "us-east1-c"
+ *   source_image = "bloxone-v381"
+ *
+ *   network_interfaces = [
+ *     {
+ *       network          = "gcp-external-network"
+ *       subnetwork       = "gcp-external-subnet"
+ *       assign_public_ip = true
+ *     },
+ *     {
+ *       network          = "gcp-internal-network"
+ *       subnetwork       = "gcp-internal-subnet"
+ *     }
+ *   ]
+ *
+ *   gcp_instance_labels = {
+ *     environment = "dev"
+ *   }
+ *
+ *   tags = {
+ *     location = "office1"
+ *   }
+ *
+ *   services = {
+ *     dhcp = "start"
+ *     dns = "start"
+ *   }
+ * }
+ * ```
+ * 
+ */
+
+resource "random_uuid" "this" {}
+
+locals {
+  join_token = var.join_token == null ? bloxone_infra_join_token.this[0].join_token : var.join_token
+  tags       = merge(
+    var.tags,
+    {
+      "tf_module_host_id" = "bloxone-host-${random_uuid.this.result}"
+    }
+  )
+}
+
+resource "bloxone_infra_join_token" "this" {
+  count = var.join_token == null ? 1 : 0
+  name  = "jt-${random_uuid.this.result}"
+}
+
+resource "google_compute_instance" "this" {
+  name                = var.name
+  project             = var.project
+  zone                = var.zone
+  machine_type        = var.machine_type
+  labels              = var.gcp_instance_labels
+  deletion_protection = var.deletion_protection
+
+  boot_disk {
+    initialize_params {
+      image  = var.source_image
```

Comment:
you might also need a parameter for size. The image comes with only 59 GB, and the BloxOne documentation mentions to resize to 750 GB. 

https://docs.infoblox.com/space/BloxOneInfrastructure/350027851/GCP+Deployment#Resizing-the-disk-capacity

Reply:
added variable ( with default value and validation )